trin-core/src/portalnet/types/messages.rs

```diff
@@ -19,6 +19,66 @@ use crate::portalnet::{types::uint::U256, Enr};
 
 pub type ByteList = VariableList<u8, typenum::U2048>;
 
+/// Custom payload element of Ping and Pong overlay messages
+#[derive(Debug, PartialEq, Clone)]
+pub struct CustomPayload {
+    payload: ByteList,
+}
```

Comment:
Not sure either. Using it as a newtype and implementing `Deref` is considered an anti-pattern for some of the Rust community: 
https://stackoverflow.com/questions/45086595/is-it-considered-a-bad-practice-to-implement-deref-for-newtypes

So I prefer to leave it as it is for now and use `self.payload` instead of `self.0`...